### PR TITLE
Refactoring Sitewide Banner to Include CTA popover rendering

### PR DESCRIPTION
**Pull request by lindsayrmaher**
### What's this PR do?

This pull request refactors the `SiteWideBanner` and `CompanyPage` to merge the checks for displaying the hellobar vs the ctaPopover on certain pages. This will make it easier for us to check that one is displayed over the other within the same file and test for all expected outcomes from the same cypress test suite.

The `CtaPopover` is also rendered within the `GeneralPage` component based on an `additionalContent` prop. This PR doesn't account for removing that yet, we're waiting to confirm what pathnames that relates to and can add those to our config file once a list is finalized. 

### How should this be reviewed?

I'd love some thoughts on the best way to approach the cypress tests. Should all tests related to Hellobar and CtaPopover be handled in the sitewide-banner test suite? Or should they continue to live in the files they started in?

Desktop:
Kapture 2020-10-21 at 11 29 05

Mobile:
Kapture 2020-10-21 at 11 30 30

### Any background context you want to provide?

We'll need to update the documentation from #2404 once we turn the feature flag off. I think this will be one of two PR's for this work, with the follow up documenting and possibly consolidating cypress test files. 

### Relevant tickets

References [Pivotal #175211087](https://www.pivotaltracker.com/story/show/175211087).

### Checklist

- [x] This PR has been added to the relevant Pivotal card.
- [ ] Documentation added for new features/changed endpoints.
- [x] Added screenshots of front-end changes on small, medium, and large screens.
- [ ] Added appropriate feature/unit tests.

### CODE REVIEW

File: resources/assets/components/pages/CompanyPage/CompanyPage.js [hunk 1]
```diff
@@ -54,25 +50,6 @@ const CompanyPageTemplate = props => {
...
-                  content="Want to earn easy scholarships for volunteering?
-                Subscribe to DoSomething's monthly scholarship email."
-                  handleClose={handleClose}
-                >
-                  <CtaPopoverEmailForm handleComplete={handleComplete} />
-                </CtaPopover>
-              </DelayedElement>
-            )}
-          />
-        ) : null}
```
Comment (aaronschachter) [inline resources/assets/components/pages/CompanyPage/CompanyPage.js:75]: We'll still want to display this if the feature flag is turned off. I was thinking about using the feature flag for incrementally making the changes -- but they seem to all be here. I think we could just remove the feature flag entirely, and avoid merging until we've confirmed the list of paths?

File: resources/assets/components/utilities/SitewideBanner/config.js [hunk 2]
```diff
@@ -14,4 +14,13 @@ export const excludedPaths = [
   '/us/quiz-results/*',
 ];
 
-export default excludedPaths;
+/**
+ * Paths to check for when we want to display the newsletter popover
+ */
+export const newsLetterWidgetPaths = [
```
Comment (aaronschachter) [inline resources/assets/components/utilities/SitewideBanner/config.js:20]: Let's define this as `scholarshipsNewsletterWidgetPaths`. We'll have different paths for [newsletter widgets for different topics ](https://github.com/DoSomething/phoenix-next/blob/master/resources/assets/components/pages/AccountPage/Subscriptions/EmailSubscriptions.js#L9-L37)(e.g. https://www.pivotaltracker.com/n/projects/2401401/stories/175377416 is for a different `lifestyle` subscription topic)

**Review by aaronschachter — COMMENTED** [review]
Re: tests, I think the cleanest approach would be to move the majority of the tests in [company-page which test the functionality of the newsletter](https://github.com/DoSomething/phoenix-next/blob/master/cypress/integration/company-page.js#L24-L83) into a separate `cta-popover` test. The CTA popover is no longer tied to a company/general/whatever page and can be used anywhere. The sitewide banner test could potentially only test the visibility of the newsletter on a page both defined and not defined within the config, but it shouldn't be responsible for testing the functionality of the widget.

File: config/feature-flags.php [hunk 3]
```diff
@@ -18,9 +18,9 @@
     'dynamic_explore_campaigns' => env('DS_ENABLE_DYNAMIC_CAMPAIGNS_PAGE', false),
     'hide_campaigns' => env('DS_ENABLE_HIDE_CAMPAIGNS', false),
     'nps_survey' => env('DS_ENABLE_NPS_SURVEY', false),
+    'post_confirmation_page' => env('DS_ENABLE_POST_CONFIRMATION_PAGE', false),
```
Comment (lindsayrmaher) [inline config/feature-flags.php:21]: there was a note to list these alphabetically and this guy was out of order when i added our short lived feature flag!

File: cypress/integration/company-page.js [hunk 4]
```diff
@@ -42,7 +44,9 @@ describe('Company Page', () => {
       },
     });
 
-    cy.visit(`/us/about/easy-scholarships`);
+    cy.withFeatureFlags({
+      sitewide_popover_cta: true,
+    }).visit(`/us/about/easy-scholarships`);
```
Comment (aaronschachter) [inline cypress/integration/company-page.js:49]:
```suggestion
    cy.visit(`/us/about/easy-scholarships`);
```

File: cypress/integration/company-page.js [hunk 5]
```diff
@@ -66,7 +70,9 @@ describe('Company Page', () => {
...
```
Comment (aaronschachter) [inline cypress/integration/company-page.js:75]:
```suggestion
    cy.visit(`/us/about/easy-scholarships`);
```

File: resources/assets/components/utilities/SitewideBanner/config.js [hunk 6]
```diff
@@ -14,4 +14,13 @@ export const excludedPaths = [
...
 
-export default excludedPaths;
+/**
+ * Paths to check for when we want to display the newsletter popover
+ */
+export const scholarshipsNewsletterPaths = [
+  '/us/about/easy-scholarships',
+  '/us/articles/how-to-apply-for-scholarships-like-a-pro',
+  '/us/articles/6-free-places-to-find-no-essay-scholarships',
+  '/us/articles/10-places-to-find-scholarships-for-high-school-seniors',
```
Comment (aaronschachter) [inline resources/assets/components/utilities/SitewideBanner/config.js:24]: Let's list these in alphabetical order to help keep this maintainable (move these line before the `/articles/how-`)

**Review by aaronschachter — COMMENTED** [review]
The [scholarship-page](https://github.com/DoSomething/phoenix-next/blob/master/cypress/integration/scholarship-page.js) test is failing in Cypress. As we chatted about in our call today, I think it's safe to move this test entirely, as it's only checking for text -- and our other tests should cover the changes we're making here (the main changes being to DRY the newsletter CTA, and use path instead of additionalContent)

**Review by aaronschachter — APPROVED** [review]
This rules! Great DRY. 🎸

**Pull request merged**